src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -569,9 +569,11 @@ int get_num_of_asic() {
                     str_num_asic = strtok_r(NULL, "=", &saveptr);
                     strip_trailing_newline(str_num_asic);
                     if (str_num_asic != NULL){
-                        sscanf(str_num_asic, "%d",&num_asic);
+                        num_asic = strtol(str_num_asic, NULL, 10);
+                        if (num_asic > 0) {
+                            break;
```

Comment:
I did not understand the usage of break here?
If num_asic is not greater than 0, then what might happen?

In what scenario will this check be useful? 
Previously, we were breaking after parsing the line starting with "NUM_ASIC=", we could still do the same without check for the value?